eZ/Publish/Core/Persistence/Cache/ObjectStateHandler.php

```diff
@@ -32,85 +34,103 @@ public function createGroup(InputStruct $input)
+
+                return $this->persistenceHandler->objectStateHandler()->loadGroupByIdentifier($identifier);
+            },
+            static function (Group $group): array {
+                return ['state-group-' . $group->id];
+            },
+            static function () use ($identifier): array {
+                return ['ez-state-group-' . $identifier . '-by-identifier'];
+            },
+            '-by-identifier'
+        );
     }
 
     /**
      * {@inheritdoc}
      */
     public function loadAllGroups($offset = 0, $limit = -1)
     {
-        $cacheItem = $this->cache->getItem('ez-state-group-all');
-        if ($cacheItem->isHit()) {
-            return array_slice($cacheItem->get(), $offset, $limit > -1 ? $limit : null);
-        }
-
-        $this->logger->logCall(__METHOD__, ['offset' => $offset, 'limit' => $limit]);
-        $stateGroups = $this->persistenceHandler->objectStateHandler()->loadAllGroups(0, -1);
-
-        $cacheItem->set($stateGroups);
-        $cacheTags = [];
-        foreach ($stateGroups as $group) {
-            $cacheTags[] = 'state-group-' . $group->id;
-        }
-        $cacheItem->tag($cacheTags);
-        $this->cache->save($cacheItem);
-
-        return array_slice($stateGroups, $offset, $limit > -1 ? $limit : null);
+        $stateGroups = $this->getCacheValue(
+            '',
+            'ez-state-group-all',
+            function () use ($offset, $limit): array {
+                $this->logger->logCall(__METHOD__, ['offset' => (int) $offset, 'limit' => (int) $limit]);
+
+                return $this->persistenceHandler->objectStateHandler()->loadAllGroups(0, -1);
+            },
+            static function (array $stateGroups): array {
+                $cacheTags = [];
+                foreach ($stateGroups as $group) {
+                    $cacheTags[] = 'state-group-' . $group->id;
+                }
+
+                return $cacheTags;
+            },
+            static function (): array {
+                return ['ez-state-group-all'];
+            }
+        );
+
+        return \array_slice($stateGroups, $offset, $limit > -1 ? $limit : null);
     }
 
     /**
      * {@inheritdoc}
      */
     public function loadObjectStates($groupId)
     {
-        $cacheItem = $this->cache->getItem('ez-state-list-by-group-' . $groupId);
-        if ($cacheItem->isHit()) {
-            return $cacheItem->get();
-        }
-
-        $this->logger->logCall(__METHOD__, ['groupId' => $groupId]);
-        $objectStates = $this->persistenceHandler->objectStateHandler()->loadObjectStates($groupId);
-
-        $cacheItem->set($objectStates);
-        $cacheTags = ['state-group-' . $groupId];
-        foreach ($objectStates as $state) {
-            $cacheTags[] = 'state-' . $state->id;
-        }
-        $cacheItem->tag($cacheTags);
-        $this->cache->save($cacheItem);
+        $objectStates = $this->getCacheValue(
```

Comment:
Static analysis: this variable accumulates memory for no reason. You can simply return the result.